src/shared/utils/arrowUpAutoFill.ts

```diff
@@ -0,0 +1,28 @@
+export const arrowUpAutoFill = () => {
+    window.onload = () => {
+        const $inputBox = document.getElementById("prompt-textarea");
+        let $lastMessage: string | null = "";
+
+        const updateDivs = () => {
+            const $divs = document.querySelectorAll(
+                'div[data-message-author-role="user"]',
+            );
+            $lastMessage = $divs[$divs.length - 1].firstChild.textContent;
+        };
+
+        updateDivs();
+        setInterval(updateDivs, 1000);
+
+        if (
+            $lastMessage &&
+            $inputBox === document.activeElement &&
+            $inputBox.textContent === ""
+        ) {
+            document.onkeydown = (e) => {
+                if (e.key === "ArrowUp") {
+                    $inputBox.textContent = $lastMessage;
+                }
+            };
+        }
+    };
+};
```

Comment:
I think a better solution would be to query the element that holds the last user message in the event handler (so when the up arrow is pressed). That way, we don't need to make a query every second and can get rid of the `setInterval`.
We might also want to add the event listener to the input box instead of the document.

I'm thinking to just have an event handler function that 
1. Checks if the up arrow was pressed. If so
    1. Queries the last user message element that contains the text content.
    2. Reassigns the input box's **value** to the queried element's **text content**.